### PR TITLE
Update docs w 1.0.0 breaking changes + sidebar tweaks + stop committing generated api docs

### PR DESCRIPTION
**Pull request by mongodben**
update docs to reflect the breaking changes part of 1.0.0 release. 

also: 

1. expand info about using stage tags without specifying a `--state` flag
1. tweaked sidebar to make sections expanded by default 
1. added  note about generated API docs to .gitignore

### CODE REVIEW

File: docs/docs/develop/api/interfaces/ListTagArgs.md [hunk 1]
```diff
@@ -34,7 +34,7 @@ ___
 
 #### Defined in
 
-[src/bluehawk/actions/ActionArgs.ts:3](https://github.com/dacharyc/Bluehawk/blob/2b37a07/src/bluehawk/actions/ActionArgs.ts#L3)
+[src/bluehawk/actions/ActionArgs.ts:3](https://github.com/mongodben/Bluehawk/blob/be77c09/src/bluehawk/actions/ActionArgs.ts#L3)
```
Comment (nathan-contino-mongo) [inline docs/docs/develop/api/interfaces/ListTagArgs.md:37]: Why doesn't this point to the mongodb-university Bluehawk repo?

File: docs/docs/reference/tags.md [hunk 2]
```diff
@@ -86,9 +86,8 @@ must appear on the same line as the tag itself.
 
 ## Snippet
 
-The `snippet` tag, also aliased as `code-block`, marks a range of content in a file
-as a snippet. You can use the [snip](#snip) CLI command to generate snippet files from
-these snippets.
+The `snippet` tag marks a range of content in a file as a snippet.
```
Comment (nathan-contino-mongo) [inline docs/docs/reference/tags.md]:
```suggestion
> 💡 Prior to version 1.0, bluehawk accepted `code-block` as an alias for `snippet`. Version 1.0 removed the `code-block` alias.

The `snippet` tag marks a range of content in a file as a snippet.
```

Could we consider a warning/info admonition to let folks know that we removed the alias in 1.0? Not sure how widespread use might be, but this seems like fair due diligence (which we can remove in the next major release).

**Review by nathan-contino-mongo — APPROVED** [review]
LGTM, two questions.

File: docs/docs/develop/api/interfaces/ListTagArgs.md (hunk 1 above)
Comment (mongodben) [reply docs/docs/develop/api/interfaces/ListTagArgs.md:37]: i don't know, but it doesn't matter b/c regenerated when built for prod. but yes it's weird. i tried to just .gitignore all the generated files, but that makes them not generate, so i think we stuck with this.

**Pull request merged**